xxhash.c

```diff
@@ -993,7 +1191,7 @@ XXH_PUBLIC_API unsigned long long XXH64_digest (const XXH64_state_t* state)
 
     h64 += (U64) state->total_len;
 
-    return XXH64_finalize(h64, state->mem64, (size_t)state->total_len, XXH_aligned);
+    return XXH64_finalize(h64, state->mem64, (size_t)state->memsize, XXH_aligned);
```

Comment:
There's a need for consistency between the 32 and 64 bit versions.
Either select to use the total `len` when invoking `_finalize()`, and expect `_finalize()` to do the final masking,
or select to use the `len` of the final partial stripe, and `assert()` that it's correctly set in the `_finalize()`.

The first one looks more general, because it doesn't add constraint to the caller, but the second one might saving one masking operation (which probably does not matter much).

Either way is fine with me. Consistency matters more for this point.

Reply:
state->memsize is supposed to hold the number of bytes residing in the accumulator and since we refer to the accumulator, I find it more consistent to use the memsize parameter istead of the total_len (the size is anded after all and both numbers give the same result, but this looks somehow clearer, IMO).